src/topology/mod.rs

```diff
@@ -687,7 +687,7 @@ fn handle_errors(
         .map_err(|_| ())
         .flatten()
         .or_else(move |()| {
-            error!("Unhandled error");
+            error!("An error occured that vector can't fix");
```

Comment:
I don't see what this has to do with making encoding optional.

Reply:
Directly nothing, they were both a minor issues reported in #2778, this one is a clarification of what `unhandled` means in this context. Do you think this deserves a separate PR?

Reply:
No, I guess not, however this is still vague. I suggest passing the task typetag and name to `handle_errors` and doing something like the following:
```rust
error!(message = "An error occurred that vector couldn't handle", r#type = typetag, %name);
```

Reply:
True, that would be better.

Reply:
I've increased the scope of instrumentation span instead of passing name and typetag directly.

Reply:
Even better